astropy/modeling/core.py

```diff
@@ -1076,20 +1204,27 @@ def _initialize_parameter_values(self, params):
         total_size = 0
 
         for name in self.param_names:
+            param_descr = getattr(self, name)
+
             if params.get(name) is None:
-                default = getattr(self, name).default
+                default = param_descr.default
 
                 if default is None:
                     # No value was supplied for the parameter, and the
-                    # parameter does not have a default--therefor the model is
+                    # parameter does not have a default--therefore the model is
                     # underspecified
                     raise TypeError(
                         "{0}.__init__() requires a value for parameter "
                         "{1!r}".format(self.__class__.__name__, name))
 
                 value = params[name] = default
+                unit = param_descr.unit
             else:
                 value = params[name]
+                if isinstance(value, Quantity):
```

Comment:
Something like that might be able to work at some point, but I really want to limit this strictly to Quantity for now so that someone's not like "I passed in a _____ and it didn't work!"